ctapipe/coordinates/angular_frames.py

```diff
@@ -445,12 +197,122 @@ def telescope_to_camera(telescope_coord, camera_frame):
         x_rotated = x_pos * cos(rot) - y_pos * sin(rot)
         y_rotated = x_pos * sin(rot) + y_pos * cos(rot)
 
-    focal_length = camera_frame.focal_length
     # Remove distance units here as we are using small angle approx
-    x_rotated = x_rotated.to(u.rad) * (focal_length / u.m)
-    y_rotated = y_rotated.to(u.rad) * (focal_length / u.m)
+    x_rotated = x_rotated * focal_length
+    y_rotated = y_rotated * focal_length
+
+    return CameraFrame(x_rotated * u.m, y_rotated * u.m, **telescope_coord.copy_properties())
+
+
+class AngularCoordinate(BaseCoordinate):
+    """
+
+    """
+    system_order = np.array(["CameraFrame", "TelescopeFrame",
+                             "NominalFrame", "HorizonFrame"])
+
+    transformations = np.array([camera_to_telescope, telescope_to_nominal,
+                                nominal_to_horizon])
+    reverse_transformations = np.array([telescope_to_camera, nominal_to_telescope,
+                                        horizon_to_nominal])
+
+    def __init__(self, focal_length=None, telescope_pointing=None, array_pointing=None,
+                 rotation=0 * u.deg):
+        """
+        Parameters
+        ----------
+        focal_length: ndarray
+            Focal length of telescope
+        telescope_pointing: HorizonFrame
+            Pointing direction of telescope
+        array_pointing: HorizonFrame
+            Pointing direction of array
+        rotation: ndarray
+            Rotation angle of camera in telescope
+        """
+        self.focal_length = focal_length
+        self.telescope_pointing = telescope_pointing
+        self.array_pointing = array_pointing
+        self.rotation = rotation
+
+        prop_dict = dict()
+        for key in self.__dict__:
+            prop_dict[key] = self.__dict__[key]
+        self.properties = prop_dict
+
+        return
+
+    def copy_properties(self):
+        """
+        Create a copy of the shared class parameters to share with other classes
+
+        Returns
+        -------
+        dict: Dictionary of shared class parameters
+        """
+        properties = self.properties
+        return properties
+
+
+class CameraFrame(AngularCoordinate, Cartesian2D):
```

Comment:
How exactly is the camera frame defined? What is the definition of the x and y axis? Do we look onto the camrera from the dish or through the camera plane into the sky?
What about two mirror telescopes?

Reply:
There is a document in preperation by the project office about common coordinate systems, and I think this could be called the "ReconstructionCameraFrame" or something like that, to differentiate it from the EngineeringCameraFrame, where x is to the right and we are looking at the front of the camera. Anyhow, those can be applied later.  Eventually I would like to have a frame that at least differentiates the strange monte-carlo coordinate system and a "nice" user camera plot coordinate system, where the x-axis is horizontal. Then we load data from the MCs in the MC frame, and can do the transform in the CameraDisplay to show it correctly.

Reply:
Hi @kosack.
Can we have access to this document to start working on cleaning the coordinates or shall we wait for it to be released (where?) ?